core/arch/arm/plat-k3/main.c

```diff
@@ -87,3 +87,15 @@ TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
 
 	return TEE_SUCCESS;
 }
+
+static TEE_Result main(void)
+{
+	uint32_t swrev = 0;
+
+	if (!ti_sci_get_swrev(&swrev))
+		IMSG("Secure Software: Rev %d", swrev);
+
+	return TEE_SUCCESS;
+}
+
+boot_final(main);
```

Comment:
warning: boot final comes before drivers initcall levels when `CFG_NS_VIRTUALIZATION` is enabled.

Reply:
I'll just use service_init_late - I think it is better to to decouple it from init_ti_sci (the SWREV value is written during the board provisioning phase and might change over time). We might also be printing some other secure boot related information in the future so this seems a bit tidier IMO.

Reply:
> . We might also be printing some other secure boot related information in the future so this seems a bit tidier IMO.

just added this and extended the PR.